pkg/proxy/integrations/genericParser/util.go

```diff
@@ -24,58 +24,46 @@ func PostgresDecoder(encoded string) ([]byte, error) {
 }
 
 func fuzzymatch(tcsMocks []*models.Mock, requestBuffers [][]byte, h *hooks.Hook) (bool, []models.GenericPayload) {
-
 	for idx, mock := range tcsMocks {
 		if len(mock.Spec.GenericRequests) == len(requestBuffers) {
-			for requestIndex, reqBuff := range requestBuffers {
+			matched := true // Flag to track if all requests match
 
-				// bufStr := string(reqBuff)
-				// if !IsAsciiPrintable(bufStr) {
+			for requestIndex, reqBuff := range requestBuffers {
 				bufStr := string(reqBuff)
 				if !IsAsciiPrintable(string(reqBuff)) {
 					bufStr = base64.StdEncoding.EncodeToString(reqBuff)
 				}
-				// }
+
 				encoded := []byte(mock.Spec.GenericRequests[requestIndex].Message[0].Data)
 				if !IsAsciiPrintable(mock.Spec.GenericRequests[requestIndex].Message[0].Data) {
 					encoded, _ = PostgresDecoder(mock.Spec.GenericRequests[requestIndex].Message[0].Data)
 				}
 
-				if string(encoded) == string(reqBuff) || mock.Spec.GenericRequests[requestIndex].Message[0].Data == bufStr {
-					log.Debug("matched in first loop")
-					tcsMocks = append(tcsMocks[:idx], tcsMocks[idx+1:]...)
-					h.SetTcsMocks(tcsMocks)
-					return true, mock.Spec.GenericResponses
+				// Compare the encoded data
+				if string(encoded) != string(reqBuff) && mock.Spec.GenericRequests[requestIndex].Message[0].Data != bufStr {
```

Comment:
shouldn't this be "or" condition, even if any one condition is false that means both requests are not same right ?